pkg/trace/config/apply.go

```diff
@@ -357,7 +365,39 @@ func (c *AgentConfig) applyDatadogConfig() error {
 		GRPCPort:        grpcPort,
 		MaxRequestBytes: c.MaxRequestBytes,
 	}
+	if c.TelemetryConfig == nil {
```

Comment:
You could make sure it's never `nil` by initialising it in `New()`, along with its defaults (such as the endpoints below).

Reply:
Its initialized in New. But I've seen examples in the apply that already don't seem to trust anything being properly initialized, so I thought I'll follow suite. 

Also it seems apply mostly replaces everything either way, so New() seems kind of unnecessary :) 

I'll remove it though, as it as you point out is not strictly necessary addition